src/posix/platform/spi_interface.cpp

```diff
@@ -310,6 +311,12 @@ void SpiInterface::InitSpiDev(const char *aPath, uint8_t aMode, uint32_t aSpeed)
 
 void SpiInterface::TriggerReset(void)
 {
+    if (otSysHardwareReset() != OT_ERROR_NOT_IMPLEMENTED)
```

Comment:
The new API seems to be used only with the SPI interface and in the `SpiInterface::HardwareReset()` module.

Should it also be used generally with any interface used by the `RadioSpinel` host? For example, we may use UART and HDLC interface for `RadioSpinel`.

It may make sense to move it to common code (so it can be used regardless of what Spinel interface is being used); for example, in `SpinelDriver::ResetCoprocessor()`, where `(mSpinelInterface->HardwareReset())` is called.

Reply:
Same concern here. Unless we do have a need for `otSysHardwareReset()`, I'd suggest we don't add it. How about quitting with `OT_EXIT_RADIO_SPINEL_NO_RESPONSE` at https://github.com/openthread/openthread/blob/f6cee7950d6d4b45ebc9779024b0b456e87159df/src/lib/spinel/spinel_driver.cpp#L154 when the hardware reset failed?